test/unit/rules/no-unsupported-role-attributes-test.js

```diff
@@ -0,0 +1,233 @@
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+generateRuleTests({
+  name: 'no-unsupported-role-attributes',
+
+  config: true,
+
+  good: [
+    '<div role="button" aria-disabled="true"></div>',
+    '<div role="heading" aria-level="1" />',
+    '<span role="checkbox" aria-checked={{this.checked}}></span>',
+    '<CustomComponent role="banner" />',
+    '<div role="textbox" aria-required={{this.required}} aria-errormessage={{this.error}}></div>',
+    '<dialog />',
+    '<a href="#" aria-describedby=""></a>',
+    '<menu type="toolbar" aria-hidden="true" />',
+    '<menuitem type="command" aria-labelledby={{this.label}} />',
+    '<input type="image" aria-atomic />',
+    '<input type="submit" aria-disabled="true" />',
+    '<select aria-expanded="false" aria-controls="ctrlID" />',
+  ],
```

Comment:
Are these examples with custom/unknown elements all valid cases? In that we don't enforce anything on them?
* `<foo role="button" aria-fake="true">`
* `<foo-component role="button" aria-fake="true">`
* `<FooComponent role="button" aria-fake="true">`
* `<foo role="fake" aria-fake="true">`

Reply:
The first three cases would fail the tests since aria-fake is not a supported ARIA property for the `button` role. They will fail in a similar way as <div role="option" aria-notreal="bogus" aria-selected="false" />.
The last case should already be caught by the [no-invalid-role](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-invalid-role.md) rule.

Reply:
Got it, just want to make sure we include every variation.

Can you make sure we have test cases with the different styles of elements / custom components?

* `<foo>`
* `<FooComponent>`
* `{{foo-component}}`

Reply:
Currentlythere are good and bad test cases for elements <foo> and angle-bracket components <CustomComponent />.  However curly-component invocations {{foo}} are allowed to pass, because it could be difficult to differentiate between component arguments and attribute values in these invocations.

Reply:
Okay, I think it could still be worth adding test cases for the curly components, even if the output is wrong, and then commenting to say why these test cases are wrong / have this limitation right now. This helps make it clear what the behavior should be in each situation and even what potential improvements could be made to the rule later or at least what improvements were considered.

Reply:
Thanks! I think it's important to continue covering/testing mustache statements/components for completeness.